conda_libmamba_solver/libmamba_utils.py

```diff
@@ -0,0 +1,320 @@
+# Copyright (C) 2019, QuantStack
+# SPDX-License-Identifier: BSD-3-Clause
+
+# TODO: Temporarily vendored from mamba.utils v0.19 on 2021.12.02
+# Decide what to do with it when we split into a plugin
```

Comment:
Just making sure we're comfortable adding this at the moment, do we need anything specific from this, or all of it?

Reply:
I only kept the parts we need. If we don't ship the current `libmamba` we can remove a few more.

Reply:
Notice that we had to add some small fixes here for our own purposes, which might have to be upstreamed.

Reply:
Assuming we remember which things to upstream, I think Wolf would appreciate it.